src/main/scala/io/smartdatalake/workflow/dataobject/DeltaLakeTableDataObject.scala

```diff
@@ -0,0 +1,150 @@
+                                   (@transient implicit val instanceRegistry: InstanceRegistry)
+  extends TransactionalSparkTableDataObject with CanHandlePartitions {
+
+  /**
+   * Connection defines db, path prefix (scheme, authority, base path) and acl's in central location
+   */
+  private val connection = connectionId.map(c => getConnection[HiveTableConnection](c))
+
+  // prepare final path and table
+  @transient private[workflow] lazy val hadoopPath = HdfsUtil.prefixHadoopPath(path, connection.map(_.pathPrefix))
+  @transient private var filesystemHolder: FileSystem = _
+  def filesystem(implicit session: SparkSession): FileSystem = {
+    if (filesystemHolder == null) {
+      filesystemHolder = HdfsUtil.getHadoopFsFromSpark(hadoopPath)
+    }
+    filesystemHolder
+  }
+  table = table.overrideDb(connection.map(_.db))
+  if (table.db.isEmpty) {
+    throw ConfigurationException(s"($id) db is not defined in table and connection for dataObject.")
+  }
+
+  override def prepare(implicit session: SparkSession): Unit = {
+    require(isDbExisting, s"($id) DB ${table.db.get} doesn't exist (needs to be created manually).")
+  }
+
+  override def getDataFrame(partitionValues: Seq[PartitionValues] = Seq())(implicit session: SparkSession): DataFrame = {
+    val df = session.read.format("delta").load(hadoopPath.toString)
+    validateSchemaMin(df)
+    df
+  }
+
+  override def init(df: DataFrame, partitionValues: Seq[PartitionValues])(implicit session: SparkSession): Unit = {
+    // create tables if possible
+    prepare(session)
+    if (!isTableExisting) {
+      logger.info(s"($id) Creating table ${table.fullName}.")
+      writeDataFrame(df, createTableOnly = true, partitionValues)
+    }
+  }
+
+  override def writeDataFrame(df: DataFrame, partitionValues: Seq[PartitionValues])
+                             (implicit session: SparkSession): Unit = {
+    validateSchemaMin(df)
+    writeDataFrame(df, createTableOnly=false, partitionValues)
+  }
+
+  /**
+   * Writes DataFrame to HDFS/Parquet and creates Hive table.
+   * DataFrames are repartitioned in order not to write too many small files
+   * or only a few HDFS files that are too large.
+   */
+  def writeDataFrame(df: DataFrame, createTableOnly: Boolean, partitionValues: Seq[PartitionValues])
+                    (implicit session: SparkSession): Unit = {
+    val dfPrepared = if (createTableOnly) {
+      // create empty df with existing df's schema
+      session.createDataFrame(List.empty[Row].asJava, df.schema)
+    } else {
+      // pass DataFrame straight through if numInitialHdfsPartitions == -1, in this case the file size in the responsibility of the framework and must be controlled in custom transformations
+      if(numInitialHdfsPartitions == -1) df
+      // estimate number of partitions from existing data, otherwise use numInitialHdfsPartitions
+      else if (isTableExisting) {
```

Comment:
We should check if path is existing... see Hadoop/SparkFileDataObject

Reply:
Done: https://github.com/smart-data-lake/smart-data-lake/pull/13/commits/b697cc8e37f9e6ef0ea576577d4130757e65535a